firmware/src/hal/sgx/test/common/common.mk

```diff
@@ -20,17 +20,24 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-SRCDIR = ../../src/trusted
+HALSRCDIR = ../../src/trusted/
+SGXSRCDIR = ../../../../sgx/src/
+TRUSTEDSRCDIR = ../../../../sgx/src/trusted/
+COMMONSRCDIR = ../../../../common/src/
+POWHSMSRCDIR = ../../../../powhsm/src/
 MOCKDIR = ../mock
 HALINCDIR = ../../../../hal/include
 TESTCOMMONDIR = ../common
 CFLAGS = -Wall -Wextra -Werror -Wno-unused-parameter -Wno-unused-function
-CFLAGS += -iquote $(SRCDIR) -iquote $(HALINCDIR)
+CFLAGS += -iquote $(HALSRCDIR) -iquote $(HALINCDIR)
 CFLAGS += -iquote $(TESTCOMMONDIR)
+CFLAGS += -iquote $(POWHSMSRCDIR)
+CFLAGS += -iquote $(COMMONSRCDIR)
+CFLAGS += -iquote $(SGXSRCDIR)
 CFLAGS += -I$(MOCKDIR)
 CFLAGS += -DHSM_PLATFORM_SGX
 
-VPATH += $(MOCKDIR):$(SRCDIR)
+VPATH += $(MOCKDIR):$(HALSRCDIR):$(TRUSTEDSRCDIR)
```

Comment:
What do we need the `HALSRCDIR` for here?

Reply:
That resolves to `firmware/src/hal/sgx/src/trusted`, where all the other hal modules are located (nvmem, secret_store, seed, etc). I just renamed this from `SRCDIR` to `HALSRCDIR` to differentiate from `SGXSRCDIR`, which is where `system.c` is located.

Reply:
I see, my bad. I've literally just noticed after this comment that you put the unit tests under `firmware/src/hal/sgx/test`, when in reality they belong in (the currently nonexistent) `firmware/src/sgx/test` (since the module you are testing is under `firmware/src/sgx/src`).

Reply:
Sorry, my bad. Yes, it makes sense, there's no point in this being under `hal`, will fix this.

Reply:
Updated to the correct location